### PR TITLE
Tu1ly/swagger json

### PR DESCRIPTION
**Pull request by tu1ly**
https://github.com/apiaryio/apiary-client/issues/137

### CODE REVIEW

**Comment by tu1ly** [comment]
[Apiary CLI should support JSON files (?)](https://trello.com/c/ww18aXRz/194-apiary-cli-should-support-json-files)

**Comment by JackuB** [comment]
I was thinking that there could also be a warning if you try to use JSON without `--json`

**Comment by tu1ly** [comment]
@JackuB  yup  - that was the idea but then i realized that we need more complex approach for that - detect format in client. it would be confusing otherwise
- no --json option
  - invalid yaml passed - `Apiary service responded with an error`  || `invalid yaml client message` (?) 
  - invalid bp passed - `Apiary service responded with an error`
  - invalid json passed - `Apiary service responded with an error` || `invalid json client message` (?) 
 - valid json passed -  `no  json param client message` (?)

**Comment by tu1ly** [comment]
+ valid json/yaml but not valid as ADD

**Comment by JackuB** [comment]
@tu1ly my point is that, if you forget `--json`, and you will get a parsing error back - can you also get `You forgot the --json flag?` kind of message?

**Comment by tu1ly** [comment]
@JackuB added - but i don't see any added value

File: lib/apiary/command/preview.rb [hunk 1]
```diff
@@ -90,10 +91,20 @@ def write_generated_path(path, outfile)
 
     def generate
       template = load_preview_template
+      source = api_description_source(@source_path)
+
+      return if source.nil?
+
+      begin
+        JSON.parse(source)
+        abort('Did You forgot the --json flag?') unless @options.json
```
Comment (JackuB) [inline lib/apiary/command/preview.rb]: *forget
capitalized You

**Comment by JackuB** [comment]
It could help with user experience - easy to forget these flags

File: lib/apiary/command/preview.rb (hunk 1 above)
Comment (tu1ly) [inline lib/apiary/command/preview.rb]: fixed

**Pull request merged**